src/nupic/math/SparseBinaryMatrix.hpp

```diff
@@ -2002,7 +2002,7 @@ namespace nupic {
 	  << "Invalid index: " << *it
 	  << " - Should be >= 0 and < number of columns:" << nCols();
       
-      for (size_type i = 1; i < (size_type)(end - begin); ++i)
+      for (auto i = 1; i < (end - begin); ++i)
```

Comment:
Hmm, not clear to me what the issue is but I think it may be that you additionally need to cast `begin` or `end` to the same type before using `+` operator. That may be necessary because `size_type` is an alias of a template parameter, although I'm not sure why.